src/segmentAnalytics.js

```diff
@@ -1,20 +1,105 @@
-export const recordPageVisit = (walletAddress = undefined) => {
-  if (window?.analytics?.page) {
-    window.analytics.page();
-  } else {
-    console.error("Tried to record page but analytics.page is undefined");
-  }
-  if (walletAddress) {
-    identifyUser(walletAddress);
-  }
-}
-
-export const identifyUser = (walletAddress) => {
-  if (window?.analytics?.identify) {
-    window.analytics.identify(walletAddress, {
-      walletAddress
-    });
-  } else {
-    console.error("Tried to record identify but analytics.identify is undefined");
-  }
-}
\ No newline at end of file
+import React, { createContext, useEffect, useState } from "react";
+import { AnalyticsBrowser } from "@segment/analytics-next";
+import { useLocation } from "react-router-dom";
+import { ARBITRUM, ARBITRUM_TESTNET, AVALANCHE, CURRENT_PROVIDER_LOCALSTORAGE_KEY, hasUserConsented } from "./Helpers";
+import { useWeb3React } from "@web3-react/core";
+
+const writeKey = process.env.REACT_APP_SEGMENT_WRITE_KEY;
+
+const networkName = {
+  [ARBITRUM]: "Arbitrum",
+  [ARBITRUM_TESTNET]: "Rinkeby",
+  [AVALANCHE]: "Avalanche",
+};
+
+const IGNORE_IP_CONTEXT = {
+  context: {
+    ip: 0,
+  },
+};
+
+const useValues = () => {
+  const { account } = useWeb3React();
+  const location = useLocation();
+  const [analytics, setAnalytics] = useState(undefined);
+
+  const trackLogin = (chainId, gmxBalances, balanceEth) => {
+    const hasConsented = hasUserConsented();
+    try {
+      const provider = localStorage.getItem(CURRENT_PROVIDER_LOCALSTORAGE_KEY);
+      const traits = {
+        walletProvider: provider,
+        walletAddress: account,
+        network: networkName[chainId],
```

Comment:
Not necessary but will be undefined if networkName does not have an entry for that chainId 
```suggestion
        network: networkName[chainId] ?? `Unsupported (${chainId})`,
```